checker/bin/wpi-many.sh

```diff
@@ -32,6 +35,14 @@ done
 # Make $@ be the arguments that should be passed to dljc.
 shift $(( OPTIND - 1 ))
 
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+SCRIPTPATH="${SCRIPTDIR}/wpi-many.sh"
+
+# report line numbers when the script fails, for debugging in CI, from
```

Comment:
I'm not sure how useful this is in CI.  In CI, you have the commit ID that is being used and so the line number is sufficient.  I don't object to this per se, but every bit of added complexity is an opportunity for problems or unexpected interactions.  I think it would be better to focus effort on making the error messages easier to understand.

Reply:
The comment about CI was perhaps misleading. I found this useful while debugging locally, and I think it will also be useful when we encounter failures in CI (but I don't have any evidence of that). I changed the comment to just say "useful for debugging" without specifically mentioning CI.

I don't understand this comment:
> In CI, you have the commit ID that is being used and so the line number is sufficient

The code here (or something else that uses `$LINENO`) is necessary to get the line number, so I don't see how the above fact (which is true) is relevant.

> I think it would be better to focus effort on making the error messages easier to understand.

This trap is part of that effort. Before this PR, the script ignored failures. Adding `set -eo pipefail` makes errors stop the script, but doesn't tell us where the error occurred. So, for example, we might get an error like "bad substitution" (this example from the `${*@Q}` issue we discussed briefly this morning), but the error message wouldn't include a line number. I added this trap code specifically so that errors like that also include a line number, so that they're easier to debug.

> every bit of added complexity is an opportunity for problems or unexpected interactions

That's fair, and I'll admit that this is a moderately complex debugging feature relative to the rest of the script. I found it very useful, though, since when a bash script fails the line number isn't printed automatically.

Reply:
OK, I'm fine with keeping it.  I agree it isn't very much complexity, especially if you have found it helpful while debugging.

> Adding set -eo pipefail makes errors stop the script, but doesn't tell us where the error occurred.

This is the aspect of the scripts that I was proposing should be improved.  That could still be done in addition to the trap.  (And I wasn't sure whether the trap might have interfered with other output or other behavior.)

Reply:
> This is the aspect of the scripts that I was proposing should be improved. That could still be done in addition to the trap

I'm not sure what the antecedent of "this" and "that" are in this sentence. This PR both adds `set -eo pipefail` and adds the trap. These two things have different functions:
* `set -eo pipefail` stops the script when an error occurs
* the trap intercepts the error from the above, prints a line number and some other context, and then proceeds to stop the script

Reply:
The failure I was seeing came from an invocation of `wpi.sh` by `wpi-many.sh`.  I didn't see any useful diagnostics from `wpi.sh` in that run.  I'm pointing out that it would have been better for `wpi.sh` to report a clear error message.  The logic in `wpi-many.sh` is useful as a stopgap when `wpi.sh` fails in an obscure way.

Reply:
Thanks, I understand now.

I think the changes to the `build.gradle` script that print the `wpi-out` file (which contains the stdout/stderr from `wpi.sh`) should make it easier to understand that kind of failure in the future.

Reply:
Great, thanks.